github_connector/models/github.py

```diff
@@ -0,0 +1,112 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2016-Today: Odoo Community Association (OCA)
+# @author: Sylvain LE GAL (https://twitter.com/legalsylvain)
+# @author: Sébastien BEAU <sebastien.beau@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+
+from openerp import exceptions, _
+
+from requests.auth import HTTPBasicAuth
+import requests
+import json
+import logging
+
+_logger = logging.getLogger(__name__)
+
+_MAX_NUMBER_REQUEST = 30
+
+_BASE_URL = 'https://api.github.com/'
+
+_GITHUB_URL = 'https://github.com/'
+
+_GITHUB_TYPE = [
+    ('organization', 'Organization'),
+    ('repository', 'Repository'),
+    ('user', 'User'),
+]
+
+_GITHUB_TYPE_URL = {
+    'organization': {'url': 'orgs/%s'},
+    'user': {'url': 'users/%s', 'url_by_id': 'user/%s'},
+    'repository': {'url': 'repos/%s', 'url_by_id': 'repositories/%s'},
+    'team': {'url_by_id': 'teams/%s'},
+    'issue': {'url': 'repos/%s/issues/%s'},
+    'organization_members': {'url': 'orgs/%s/members'},
+    'organization_repositories': {'url': 'orgs/%s/repos'},
+    'organization_teams': {'url': 'orgs/%s/teams'},
+    'team_members': {'url': 'teams/%s/members'},
+    'repository_issues': {'url': 'repos/%s/issues?state=all'},
+    'issue_comments': {'url': 'repos/%s/issues/%s/comments'},
+    'repository_branches': {'url': 'repos/%s/branches'},
+}
+
+
+class Github(object):
+
+    def __init__(self, github_type, login, password, max_try):
+        super(Github, self).__init__()
+        self.github_type = github_type
+        self.login = login
+        self.password = password
+        self.max_try = max_try
+
+    def list(self, arguments):
+        page = 1
+        datas = []
+        while True:
+            pending_datas = self.get(arguments, False, page)
+            datas += pending_datas
+            if pending_datas == [] or\
+                    len(pending_datas) < _MAX_NUMBER_REQUEST:
+                break
+            page += 1
+        return datas
```

Comment:
IMO this method should be a generator, with a `yield pending_datas` inside of the loop instead of the explicit list building. Unless maybe I'm missing something, but it looks like it's only iterated.